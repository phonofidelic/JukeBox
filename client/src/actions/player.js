import { 
	PLAY_TRACK,
	PAUSE_TRACK,
	STOP_TRACK,
	PLAY_NEXT,
	PLAY_PREV
} from '../actiontypes';

export const playTrack = (queue, queueIndex) => {
	queue[queueIndex].howl.play();
	return dispatch => {
		dispatch({
			type: PLAY_TRACK
		});
	};
}

export const pauseTrack = (queue, queueIndex) => {
	queue[queueIndex].howl.pause();
	return dispatch => {
		dispatch({
			type: PAUSE_TRACK
		});
	};
}

export const stopTrack = (queue, queueIndex) => {
	queue[queueIndex].howl.stop();
	return dispatch => {
		dispatch({
			type: STOP_TRACK
		});
	};
}

export const playNext = (queue, queueIndex) => {
	console.log('playNext, queueIndex:', queueIndex+1);
	queue[queueIndex].howl.stop();
	queue[queueIndex+1].howl.play();
	return dispatch => {
		dispatch({
			type: PLAY_NEXT
		})
	}
}

export const playPrev = (queue, queueIndex) => {
	console.log('playPrev, queueIndex:', queueIndex-1);
	queue[queueIndex].howl.stop();
	queue[queueIndex-1].howl.play();
	return dispatch => {
		dispatch({
			type: PLAY_PREV
		})
	}
}