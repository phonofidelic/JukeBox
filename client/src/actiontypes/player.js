export const PLAY_TRACK = 'play_track',
						 PAUSE_TRACK = 'pause_track',
						 STOP_TRACK = 'stop_track';