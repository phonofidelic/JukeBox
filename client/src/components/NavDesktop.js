import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  BottomNavigation, 
  BottomNavigationAction,
  Grid,
  Typography,
} from '@material-ui/core';
import { 
  Home,
  FileUpload,
  Storage,
  ExitToApp,
} from '@material-ui/icons';
import { withTheme } from '@material-ui/core/styles';

class NavDesktop extends Component {
  render() {
    const { 
      locationPathname,
      handleSignOut,
      theme, 
    } = this.props;

    const styles = {
      root: {
        position: 'fixed',
        left: '0px',
        top: '0px',
        height: '100vh',
        width: theme.dimensions.navDesktop.navWidth,
        padding: 10,
        backgroundColor: theme.palette.secondary.light,
        zIndex: 1,
      },
    }

    return(
      <Grid 
        container
        component={BottomNavigation} 
        value={locationPathname}
        alignContent={'flex-start'}
        style={styles.root}
        showLabels
      >
        <BottomNavigationAction 
          component={Link} 
          to="/"
          label={<Typography color="inherit" variant="caption">Home</Typography>}
          value="/" 
          icon={<Home />} 
        />
        <BottomNavigationAction 
          component={Link} 
          to="/library"
          label={<Typography color="inherit" variant="caption">Library</Typography>}
          value="/library" 
          icon={<Storage />} 
        />
        <BottomNavigationAction 
          component={Link}
          to="/uploader" 
          label={<Typography color="inherit" variant="caption">Uploader</Typography>}
          value="/uploader" 
          icon={<FileUpload />} 
        />
        <BottomNavigationAction 
          onClick={() => handleSignOut()}
          label={<Typography color="inherit" variant="caption">Sign out</Typography>} 
          icon={<ExitToApp />} 
        />
      </Grid>
    );
  };
}

NavDesktop.propTypes = {
  locationPathname: PropTypes.string.isRequired
};


export default withTheme()(NavDesktop);
