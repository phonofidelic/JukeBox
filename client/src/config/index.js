export const URLS = {
	TRACKS_URL: '/tracks'
}

export const THEME = {
	typography: {
    // Use the system font over Roboto.
    fontFamily:
      '-apple-system,system-ui,BlinkMacSystemFont,' +
      '"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif',
    fontWeightMedium: 500,
    body1: {
      fontWeight: 500,
    },
    subheading: {
      fontSize: 12,
    },
    button: {
      // fontStyle: 'italic',
      // size: 'large'
    },
  },
  palette: {
    primary: {
      main: '#d2d8d8',
      light: '#fff',
      dark: '#222'
    },
    secondary: {
      main: '#f44336',
      light: '#ff7961',
      dark: '#ba000d'
    },
    context: {
      danger: '#EF5350',
      warning: 'orange',
      success: 'green',
      info: 'blue'
    }
  },
}