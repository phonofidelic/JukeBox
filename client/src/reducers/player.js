import { 
	PLAY_TRACK,
	PAUSE_TRACK,
	STOP_TRACK,
	START_NEW_QUEUE,
	ADD_TRACK_TO_QUEUE,
	PLAY_NEXT,
	PLAY_PREV,
	TOGGLE_QUEUE_VISIBILITY,
} from '../actiontypes';

const INITIAL_STATE = {
	queue: [],
	showQueue: true,
	howl: null,
	playing: false,
	queueIndex: null,
	currentTrack: null,
	message: null
};

const player = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case START_NEW_QUEUE:
			return {
				...state,
				playing: true,
				currentTrack: action.track,
				queue: [action.track],
				queueIndex: 0
			}

		case ADD_TRACK_TO_QUEUE:
			return {
				...state,
				queue: [...state.queue, action.track],
				message: action.message
			}

		case PLAY_TRACK:
			return {
				...state,
				playing: true,
			}

		case PAUSE_TRACK:
			return {
				...state,
				playing: false,
			}

		case STOP_TRACK:
			return {
				...state,
				playing: false
			}

		case PLAY_NEXT:
			return {
				...state,
				playing: true,
				queueIndex: state.queueIndex + 1,
				currentTrack: state.queue[state.queueIndex + 1]
			}

		case PLAY_PREV:
			return {
				...state,
				playing: true,
				queueIndex: state.queueIndex - 1,
				currentTrack: state.queue[state.queueIndex - 1]
			}

		case TOGGLE_QUEUE_VISIBILITY:
			return {
				...state,
				showQueue: !state.showQueue
			}

		default: return state;
	}
};

export default player;