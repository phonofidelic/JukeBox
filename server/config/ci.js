require("dotenv").config();
module.exports = {
  // API_ROOT: 'http://localhost:3000',
  PORT: 3000,
  DB_CONNECTION: "mongodb://127.0.0.1:27017/jukebox_ci",
  CLIENT_DIR: "../client/build",
  FS_AUDIO: "uploads/audio",
  FS_IMAGE: "uploads/images",
  TMP: "tmp",
  JWT_SECRET: process.env.JWT_SECRET,
  JWT_EXP: "15m",
  JWT_AUD: "jukebox_client",
  JWT_ISS: "jukebox_api",
  DISCOGS_TOKEN: process.env.DISCOGS_TOKEN,
  // G_CLIENT_ID:
  //   '1055271812237-icje6bl0lorut3pseo494avkdn6ohgub.apps.googleusercontent.com',
  // G_CLIENT_SECRET: process.env.G_CLIENT_SECRET,
  // G_REDIRECT_URI: 'api/gdrive/authcode',
  S3_ACCESS_KEY_ID: process.env.S3_ACCESS_KEY_ID,
  S3_SECRET_ACCESS_KEY: process.env.S3_SECRET_ACCESS_KEY,
  S3_BUCKET_NAME: "jukebox-ci-storage",
  STORAGE_BASE_URL: process.env.STORAGE_BASE_URL,
  STORAGE_MAX: 16106127360
};
