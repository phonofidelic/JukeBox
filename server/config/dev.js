module.exports = {
  API_ROOT: 'http://localhost:3001',
  PORT: 3001,
  DB_CONNECTION: 'mongodb://localhost:27017/jukeTest',
  CLIENT_DIR: '../client/build',
  FS_AUDIO: 'uploads/audio',
  FS_IMAGE: 'uploads/images',
  TMP: 'tmp',
  JWT_SECRET: process.env.JWT_SECRET,
  JWT_EXP: '15m',
  JWT_AUD: 'jukebox_client',
  JWT_ISS: 'jukebox_api',
  DISCOGS_TOKEN: process.env.DISCOGS_TOKEN,
  G_CLIENT_ID: process.env.G_CLIENT_ID,
  G_CLIENT_SECRET: process.env.G_CLIENT_SECRET,
  G_REDIRECT_URI: process.env.G_REDIRECT_URI,
  S3_ACCESS_KEY_ID: process.env.S3_ACCESS_KEY_ID,
  S3_SECRET_ACCESS_KEY: process.env.S3_SECRET_ACCESS_KEY,
  S3_BUCKET_NAME: 'jukebox-dev-storage',
  STORAGE_BASE_URL: process.env.STORAGE_BASE_URL
};
