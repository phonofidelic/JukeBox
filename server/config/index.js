require('dotenv').config();
console.log('### process.env.JWT_SECRET:', process.env.JWT_SECRET)
module.exports = {
	port: process.env.PORT || 3001,
	db: process.env.MONGODB_URI || 'mongodb://localhost:27017/jukeTest',
	fileLocation_audio: './uploads/audio',
	fileLocation_images: './uploads/images',
	JWT_SECRET: process.env.JWT_SECRET	// Where did this come from???????
};