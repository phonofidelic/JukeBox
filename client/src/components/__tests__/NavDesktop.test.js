import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { createRender } from '@material-ui/core/test-utils';

import NavDesktop from '../NavDesktop';
import { INITIAL_STATE } from '../../reducers/player.reducer';
// TODO: Create mock-history instead of importing actual history config?
import { history, THEME } from '../../config';
import { ConnectedRouter } from 'react-router-redux';

/***
 *	This tests NavDesktop rendering in each of its routing states.
 *	It does not test the sign-out functionality which will probably
 *	be removed from this component to a Profile component in the 
 *	near future.
 */

describe('NavDesktop', () => {
	let props;
	let mountedNavDesktop;
	let muiRenderer;
	let mockStore;

	const navDesktop = props => {
		if (!mountedNavDesktop) {
			mountedNavDesktop = mount(
				<NavDesktop {...props} />
			);
		}
	};

	beforeEach(() => {
		props = {
			locationPathname: null,
			handleSignOut: jest.fn(),
			theme: THEME
		};

		mountedNavDesktop = undefined;
		muiRenderer = createRender();
		mockStore = { dispatch: jest.fn() };
	});

	it('renders as expected with default path', () => {
		props.locationPathname = '/';

		/***
		 *	Wrap NavDesktop in Router component to avoid errors.
		 *	Is this an anti-pattern? Test depends too much on external code? 
		 */
		const tree = renderer
		.create(
			<ConnectedRouter 
				store={mockStore} 
				history={history}
			>
				<NavDesktop {...props} />
			</ConnectedRouter>
		).toJSON();
		
		expect(tree).toMatchSnapshot();
	});

	it('renders as expected with "/library" path', () => {
		props.locationPathname = '/library';

		const tree = renderer.create(
			<ConnectedRouter
				store={mockStore}
				history={history}
			>
				<NavDesktop {...props} />
			</ConnectedRouter>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('renders as expected with "/uploader" path', () => {
		props.locationPathname = '/uploader';

		const tree = renderer.create(
			<ConnectedRouter
				store={mockStore}
				history={history}
			>
				<NavDesktop {...props} />
			</ConnectedRouter>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
