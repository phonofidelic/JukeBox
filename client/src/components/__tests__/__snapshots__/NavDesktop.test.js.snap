// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDesktop renders as expected with "/library" path 1`] = `
<div
  className="MuiBottomNavigation-root-92 MuiGrid-container-1 MuiGrid-align-content-xs-flex-start-14"
  style={
    Object {
      "backgroundColor": "#fafafa",
      "height": "100vh",
      "left": "0px",
      "padding": 10,
      "position": "fixed",
      "top": "0px",
      "width": 80,
      "zIndex": 1,
    }
  }
>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Home
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93 MuiBottomNavigationAction-selected-94"
    href="/library"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97 MuiBottomNavigationAction-selected-94"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Library
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/uploader"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Uploader
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <button
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Sign out
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </button>
</div>
`;

exports[`NavDesktop renders as expected with "/uploader" path 1`] = `
<div
  className="MuiBottomNavigation-root-92 MuiGrid-container-1 MuiGrid-align-content-xs-flex-start-14"
  style={
    Object {
      "backgroundColor": "#fafafa",
      "height": "100vh",
      "left": "0px",
      "padding": 10,
      "position": "fixed",
      "top": "0px",
      "width": 80,
      "zIndex": 1,
    }
  }
>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Home
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/library"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Library
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93 MuiBottomNavigationAction-selected-94"
    href="/uploader"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97 MuiBottomNavigationAction-selected-94"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Uploader
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <button
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Sign out
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </button>
</div>
`;

exports[`NavDesktop renders as expected with default path 1`] = `
<div
  className="MuiBottomNavigation-root-92 MuiGrid-container-1 MuiGrid-align-content-xs-flex-start-14"
  style={
    Object {
      "backgroundColor": "#fafafa",
      "height": "100vh",
      "left": "0px",
      "padding": 10,
      "position": "fixed",
      "top": "0px",
      "width": 80,
      "zIndex": 1,
    }
  }
>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93 MuiBottomNavigationAction-selected-94"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97 MuiBottomNavigationAction-selected-94"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Home
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/library"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Library
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <a
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    href="/uploader"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Uploader
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </a>
  <button
    className="MuiButtonBase-root-98 MuiBottomNavigationAction-root-93"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-96"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-101"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-97"
      >
        <span
          className="MuiTypography-root-107 MuiTypography-caption-117 MuiTypography-colorInherit-126"
        >
          Sign out
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-131"
    />
  </button>
</div>
`;
